streamz/dataframe/aggregations.py

```diff
@@ -202,16 +204,18 @@ def diff_loc(dfs, new, window=None):
         List of dataframes to decay
     """
     dfs = deque(dfs)
-    dfs.append(new)
-    mx = max(df.index.max() for df in dfs)
-    mn = mx - pd.Timedelta(window)
+    if len(new) > 0:
+        dfs.append(new)
     old = []
-    while pd.Timestamp(dfs[0].index.min()) < mn:
-        o = dfs[0].loc[:mn]
-        old.append(o)  # TODO: avoid copy if fully lost
-        dfs[0] = dfs[0].iloc[len(o):]
-        if not len(dfs[0]):
-            dfs.popleft()
+    if len(dfs) > 0:
+        mx = max(df.index.max() for df in dfs)
+        mn = mx - pd.Timedelta(window) + pd.Timedelta('1ns')
+        while pd.Timestamp(dfs[0].index.min()) < mn:
+            o = dfs[0].loc[:mn]
+            old.append(o)  # TODO: avoid copy if fully lost
```

Comment:
May as well check for an empty `old` here and fix the TODO?

Reply:
`old = old.append(o) if len(old) else o`. (+ comment to explain)

Reply:
Done